pkg/master/controller/clusterauthenticationtrust/cluster_authentication_trust_controller.go

```diff
@@ -201,6 +201,15 @@ func writeConfigMap(configMapClient corev1client.ConfigMapsGetter, required *cor
 		return err
 	}
 
+	// If the configmap is too big, clear the entire thing and count on this controller (or another one) to add the correct data back.
+	// We return the original error which causes the controller to re-queue.
+	if apierrors.IsRequestEntityTooLargeError(err) {
```

Comment:
when does this happen? Seeing this here is surprising.

Reply:
```
I1105 10:16:32.597187   11924 request.go:942] Response Body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"Request entity too large: limit is 3145728","reason":"RequestEntityTooLarge","code":413}
I1105 10:16:32.597656   11924 helpers.go:198] server response object: [{
  "metadata": {},
  "status": "Failure",
  "message": "Request entity too large: limit is 3145728",
  "reason": "RequestEntityTooLarge",
  "code": 413
}]
```

```
Response Body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"ConfigMap \"foo\" is invalid: []: Too long: must have at most 1048576 characters","reason":"Invalid","details":{"name":"foo","kind":"ConfigMap","causes":[{"reason":"FieldValueTooLong","message":"Too long: must have at most 1048576 characters","field":"[]"}]},"code":422}
F1105 10:20:18.517027   13269 helpers.go:116] The ConfigMap "foo" is invalid: []: Too long: must have at most 1048576 characters
```

https://github.com/kubernetes/kubernetes/blob/36981002246682ed7dc4de54ccc2a96c1a0cbbdb/pkg/apis/core/validation/validation.go#L5009

Reply:
This needs a (integration) test: make the configmap big such that the update hits the limits and the whole mechanism recovers. Otherwise every regression is a time bomb.